module-2/lab-probability-distribution/your_code/main.ipynb

```diff
@@ -58,10 +58,10 @@
      "name": "stdout",
```

Comment:
- Sometimes it's useful to seed the random number generator. Seeding a pseudo-random number generator gives it its first "previous" value. Each seed value will correspond to a sequence of generated values for a given random number generator. That is, if you provide the same seed twice, you get the same sequence of numbers twice. https://machinelearningmastery.com/how-to-generate-random-numbers-in-python/
- Normality tests are quite sensitive to sample size thus for a sample size below a couple of thousands it should be a better idea to evaluate normality through a plot, in particular a quantile-quantile plot (QQ plot). A Q-Q plot is a scatterplot created by plotting two sets of quantiles against one another. It plots the quantiles of our sample vs the theoretical quantiles of the distribution type we want to test (this case a normal distribution) How to do this in Python? https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.probplot.html.
- There are 2 other normality tests (additional to D'Agostino k^2 test, normaltest in Python) that are very used in to test normality of our data. These are Kolmogorov-Smirnof and Shapiro-Wilk tests. I encourage you to deepen on these type of tests. In case you are interested, I can provide you some useful resources.
- Good one!! 
```Python
plt.style.use('seaborn-white')
c = plt.cm.Dark2(4)
fig, axis = plt.subplots(1, 4, sharex=False, sharey=False, figsize=(12,3))
fig.suptitle('Normal Distributions', verticalalignment='bottom')
for i, ax in enumerate(axis): 
    axis[i].hist(normal_dists[i], bins=30, color=plt.cm.Dark2(i))
    axis[i].set_title('Sample Size = {}'.format(len(normal_dists[i])))
```
- Thanks for setting x and y label as well as title in your plots. It's a must!! 
- There's something not working quite well. I guess the problem is in the definition of x. The output of this challenge should be to plot how a binomial distribution can be approximated by a normal distribution under certain circumstances, one of them is the sample size. 
```Python
visitors = [50, 500, 5000]
p = 0.15
fig, axis = plt.subplots(1, len(visitors), sharex=False, sharey=False, figsize=(12,3))
fig.suptitle('Binomial Distributions', verticalalignment='bottom')  
for i, v in enumerate(visitors): 
    x = list(range(10+1))
    axis[i].plot(x, binom.pmf(x, n, p))
    axis[i].set_title('n = {}'.format(v))
```
- Normal distribution can be used as an approximation of Binomial and Poisson distributions under certain circumstances. Generally speaking:
Binomial: X ~ B(n, p) and if n is large and/or p is close to ½, then X is approximately N(np, npq)
Poisson: X ~ Po(l) then for large values of l, X ~ N(l, l) approximately.
However, we must be careful because binomial and poisson distributions are discrete while Normal distribution is continuous. We must use a continuity correction.